### DISCUSSION TITLE
How can I dynamically generate the file so that I can request the data file in the client？

### DISCUSSION THREAD

**Discussion by laoergege** [nuxt/content#2234] [Q&A]
I might need to collect something on markdown during build as a data piece and then look it up in the client, so how do I do that?

**Comment by TeddyHuang-00** [comment]
I think it depends on how you would like your pages being presented. If it is just a single page with fixed route then you can just do whatever you want to do with the markdown file and just build the site normally. If you want to dynamically generate route(s) then you might wanna have a look into using JSON as content source, to be specific having a list as root element in JSON will result in multiple routes being added at build time. Hope this is helpful to you.

**Reply by laoergege** [reply]
Thanks, my mistake, I found queryContent() to meet my needs

**Reply by TeddyHuang-00** [reply]
Cool!
